lib/foreman_maintain/concerns/metadata.rb

```diff
@@ -90,6 +90,12 @@ def for_feature(feature_label)
           end
         end
 
+        # Ensure to not run the step twice: expects the scenario to be presisted
```

Comment:
It's saved to data.yml, the use-case is, when there is a failure in the middle of the upgrade.